src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = Storage.getFruitStorage();
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+
+            handler.apply(storage, transaction);
+        }
+    }
+
+    @Override
+    public Map<String, Integer> getStorage() {
+        return new HashMap<>(storage);
```

Comment:
Be cautious when returning a copy of the storage map. If it's intended to prevent modification of the internal storage, this is fine. However, if the storage is meant to be a single source of truth, returning a copy could lead to inconsistencies between the returned copy and the actual storage state.